services/slam/builtin/builtin.go

```diff
@@ -481,7 +481,38 @@ func (slamSvc *builtIn) GetPosition(ctx context.Context, name string) (spatialma
 	ctx, span := trace.StartSpan(ctx, "slam::builtIn::GetPosition")
 	defer span.End()
 
-	return nil, "", errors.New("unimplemented stub")
+	req := &pb.GetPositionNewRequest{Name: name}
+
+	resp, err := slamSvc.clientAlgo.GetPositionNew(ctx, req)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "error getting SLAM position")
+	}
+	pose := spatialmath.NewPoseFromProtobuf(resp.GetPose())
+	componentReference := resp.GetComponentReference()
+	returnedExt := resp.Extra.AsMap()
+
+	return CheckQuaternionFromClientAlgo(pose, componentReference, returnedExt)
+}
+
+// CheckQuaternionFromClientAlgo checks to see if the internal SLAM algorithm sent a quaternion. If it did, return the updated pose
+func CheckQuaternionFromClientAlgo(pose spatialmath.Pose, componentReference string, returnedExt map[string]interface{}) (spatialmath.Pose, string, error) {
```

Comment:
not sure if this is the right spot for this function. Does not seem fitting to move it to internal because this function should only be used by the builtin service

Reply:
I think we should put it in internal for now. I will be removed (or changed) once we can stop using the `extra` part of the response to return the quaternion. 

More importantly it will allow us to unexpose it to outside users

Reply:
i could see putting it within a `builtin/internal/helpers.go` file, but putting it in the slam service seems incorrect to me. Current implementation means people would have to add builtin as a package in order to use it

Reply:
Let's put it into the slam repo: https://github.com/viamrobotics/slam

Can choose which package to put it into. Could drop it into utils/utils.go, or maybe create math/math.go? Up to you!

Reply:
after talking it over, going with what is currently in the PR and will migrate after map representation is complete